### PR TITLE
refactor and remove afterResolving()

### PR DESCRIPTION
**Pull request by olipayne**
This PR resolves the issue described in https://github.com/Laravel-Backpack/CRUD/issues/4251, which @preBit and I are both experiencing. 

I'm unclear on the magic going on behind the scenes when it comes to bladeCompiler, however I followed the structure defined in https://github.com/spatie/laravel-permission/blob/main/src/PermissionServiceProvider.php, and now my @loadOnce calls render properly.

### CODE REVIEW

**Comment by preBit** [comment]
I confirm this PR resolves the issue.

**Comment by pxpm** [comment]
Thanks @olipayne .

I will have a look at it later today, but if the problem is `afterResolving` I am afraid we use it in ALOT of other places, so I need to give this a throught review.

**Comment by kiddtang** [comment]
for my case. i roll back `spatie/laravel-permission (5.5.1 => 5.5.0)` solve the issue.
I am not sure it is related. just FYI https://github.com/Laravel-Backpack/CRUD/issues/4251#issuecomment-1059935166.

**Comment by olipayne** [comment]
I think @kiddtang is onto it, in `5.5.1` they refactored the blade directives code - https://github.com/spatie/laravel-permission/pull/2039/files

**Comment by preBit** [comment]
I just tested and confirm downgrading `spatie/laravel-permission (5.5.1 => 5.5.0)` and `digitallyhappy/assets at 2.0.3` without this PR solved the issue.

**Comment by olipayne** [comment]
Can confirm too, now the question is which package needs to be updated to be compatible with the other? 😬

**Comment by preBit** [comment]
Eagerly awaiting for this one...😬

**Comment by pxpm** [comment]
@olipayne , @preBit and @kiddtang 🙏 Can't thank you guys enough! Sad but true, I was using 5.5.0 so was unable to reproduce your error until I updated to 5.5.1. 

I guess that we are the ones that should update our code. We just updated from `Blade::directive` to use the `afterResolving` like two weeks ago, but from what it seems, the proper way to do it is NOT using `afterResolving`, that from my understanding should work just fine, I guess something somewhere will be fixed that will work both ways without errors, for now I think we just need to stick with the "most common practice". 

I am going to create the PRs to other packages we have that are using `afterResolving`, atmost this will be fixed tomorrow, so I guess we have a plan. Will try to make it happen today, for now use 5.0.0, since the only change in 5.5.1 was that blade directive change. 

Thanks again guys!

File: src/AssetsServiceProvider.php [hunk 1]
```diff
@@ -43,4 +44,43 @@ public function provides()
...
+        $bladeCompiler->directive('loadScriptOnce', function ($parameter) {
+            return "<?php Assets::echoJs({$parameter}); ?>";
+        });
+
+        $bladeCompiler->directive('loadOnce', function ($parameter) {
+            // determine if it's a CSS or JS file
+            $cleanParameter = Str::of($parameter)->trim("'")->trim('"')->trim('`');
+            $filePath = Str::of($cleanParameter)->before('?')->before('#');
+
+            // mey be useful to get the second parameter
```
Comment (pxpm) [inline src/AssetsServiceProvider.php:65]:
```suggestion
            // may be useful to get the second parameter
```

**Review by pxpm — APPROVED** [review]
@tabacitu just tested this. All fine, works fine. 

Note that `Impersonate` package (the one that lead us to this fix: https://github.com/Laravel-Backpack/CRUD/issues/4168 ) will stop working. They should update their code as we are updating ours, I guess since all spatie packages load directives this way, it's probably better to use this convention.

**Comment by erikn69** [comment]
https://github.com/erikn69/permission-issue/commit/5118f89811489dee0ac275af08b7d7b7339799dd 
I did test `spatie/laravel-permission 5.5.1` with this package and everything works

**Comment by tabacitu** [comment]
[Answered with steps to reproduce the scenario where it doesn't work](https://github.com/spatie/laravel-permission/pull/2047#issuecomment-1061594509). Thanks!

**Comment by tabacitu** [comment]
Let's close this in favor of https://github.com/spatie/laravel-permission/pull/2048 - I hope Freek will find the time to merge it soon 👀😅

**Comment by tabacitu** [comment]
Thanks a lot for the time you all spend on this, guys... It was one of those damned things that's difficult to pinpoint... But it's better fixed upstream.

**Comment by tabacitu** [comment]
@olipayne I ended up doing a lot of what you did... in https://github.com/DigitallyHappy/assets/pull/13 ... with 3 letters change. Sorry I didn't merge yours 🙏

**Pull request closed without merging**